docs/source/aggregation.rst

```diff
@@ -0,0 +1,172 @@
+By default, when no `sort_by` parameter is specified, the order of the jobs will be decided by the order in which the jobs are iterated in a **signac** project.
+
+.. code-block:: python
+
+    @aggregator.groupsof(2, sort_by='temperature', sort_ascending=False)
+    @Project.operation
+    def op5(job1, job2):
+        pass
+
+.. note::
+
+    In the above example, all the jobs will be sorted by the state point parameter ``temperature`` in descending order and then be aggregated as groups of 2.
+
+Selecting jobs for aggregation
+------------------------------
+
+**signac-flow** allows users to selectively choose which jobs to pass into operation functions.
+
+.. code-block:: python
+
+    @aggregator(select=lambda job: job.sp.temperature > 0)
+    @Project.operation
+    def op6(job1, job2):
+        pass
+
+
+.. _aggregate_id:
+
+Aggregate ID
+============
+
+Similar to the concept of a job id, an aggregate id is a unique hash identifying an aggregate of jobs.
+The aggregate id is sensitive to the order of the jobs in the aggregate.
+
+
+.. note::
+
+    The id of an aggregate containing one job is that job's id.
+
+In order to distinguish between aggregate id and a job id, for an aggregate of more than one job the aggregate id of that aggregate will always have a prefix ``agg-``.
+
+Users can generate the aggregate id of an aggregate using :meth:`flow.get_aggregate_id`.
+
+.. tip::
+
+    Users can also pass an aggregate id to the ``--job-id`` command-line flag provided by **signac-flow** in ``run``, ``submit``, and ``exec``.
+
+
+.. _aggregation_with_flow_groups:
+
+Aggregation with FlowGroups
+===========================
+
+In order to associate aggregator object with a :py:class:`FlowGroup`, **signac-flow** provides a ``group_aggregator`` parameter in :meth:`~flow.FlowProject.make_group`. By default, no aggregation takes place for a :py:class:`FlowGroup`.
+
+.. note::
+
+    Currently, **signac-flow** only allows single :class:`~flow.aggregator` per group, i.e., all the operations present in a :py:class:`FlowGroup` will be using the same :class:`~flow.aggregator` object.
+
+.. code-block:: python
+
+    # project.py
+    from flow import FlowProject, aggregator
+
+    class Project(FlowProject):
+        pass
+
+    group = Project.make_group('agg-group', group_aggregator=aggregator())
+
+    @group
+    @aggregator()
+    @Project.operation
+    def op1(*jobs):
+        pass
+
+    @group
+    @Project.operation
+    def op2(*jobs):
+        pass
+
+    if __name__ == '__main__':
+        Project().main()
+
+In the above example, when the group ``agg-group`` is executed, all the jobs in the project are passed as arbitrary arguments for ``op1`` and ``op2``. But if only ``op2`` is executed, only a single job is passed as a parameter.
```

Comment:
This last sentence is also a bit confusing to me (probably because I'm still not very familiar with `aggregation`). Users might be wondering if a job is randomly chosen and passed in as a parameter. I assume you mean when used alone, `op2` operates on a single-job basis, but still iterates over all jobs? (Correct me if I'm wrong)

Reply:
In the above example, `op1` have a default aggregator associated with it, `op2` doesn't have any aggregator associated with it, and the group `agg-group` (which contains both `op1` and `op2`) has a default aggregaot associated with it.
Here is the detailed explaination of how these will get executed

1. `python project.py -o agg-group` -> all jobs will get aggregated. So `op1` and `op2` will run once and the argument passed will be the aggregate of all jobs.
2. `python project.py -o op1` -> all jobs will get aggregated and `op1` will run once and the argument passed will be the aggregate of all jobs.
3. `python project.py -o op2` -> No aggregation will take place. Every job in the project will be passed as an argument individually to `op2`.

Does this explaination make the example more clear?

Reply:
Tried making it more clear here: https://github.com/glotzerlab/signac-docs/pull/122/files#diff-d053cd842dd6fe746188c4339762f7e336b763b94d513473d65900b9836a09f0R176

Reply:
Your explanation here is clear, but still in the documentation it's a bit confusing. What I was trying to get at is that the sentence:

> only a single job is passed as a parameter

is not clear enough. I think this is also related to @DomFijan's comment on adding docs for `job` vs `*jobs`